asreview/webapp/auth.py

```diff
@@ -0,0 +1,151 @@
+
+        if not auth_file:
+            self.auth_file = get_auth_file_path()
+
+        self.users = {}
+        if self.auth_file and os.path.isfile(self.auth_file):
+            try:
+                self.users = self.users_from_file
+            except Exception as e:
+                logging.error(e)
+                logging.error('Error reading auth file, no auth will be enabled.')
+
+        if not len(self.users):
+            logging.debug('No users found in auth file, no auth will be enabled.')
+            self.enabled = False
+        else:
+            self.enabled = True
+                
+
+        @self.verify_password
+        def verify_userpwd(username, password):
+            if not self.enabled:
+                return ASReviewAuth.ANONYMOUS_USERNAME
+            
+            if username in self.users and \
+                check_password_hash(self.users[username], password):
+                
+                return username
+
+    @property
+    def users_from_file(self):
+        """ Instance method of getting the users from the auth file. """
+
+        return ASReviewAuth.get_users_from_file(self.auth_file)
+
+
+    @staticmethod
+    def get_users_from_file(fp, delimiter=None):
+        """ Get the dictionary of username:hash_string in the file.
+        
+        Args:
+            fp (string): Path to the authentication file.
+            delimiter (string): String which is used to delimit user name and
+                password hash string in the file.
+
+        Returns:
+            dict: Dictionary, where keys are user names and values are the
+                hash strings of their passwords.
+        """
+        if delimiter is None:
+            delimiter = ASReviewAuth.USERPWD_DELIMITER
+
+        try:
+            with open(fp, 'r') as f:
+                users = [l.strip().split(delimiter) for l in f]
+                users = {key: value for key, value in users}
+
+            return users
+
+        except Exception as e:
+            raise Exception("Could not read the users. Check for possible corruptions " +
+                "(e.g. ensure that delimiter is '%s')." % delimiter)
+
+
+    @staticmethod
+    def write_auth_file(users, fp, delimiter=None):
+        """ Write the dictionary of user names and hash strings to the file.
+
+        Args:
+            users (dict): Dictionary, where keys are user names and values are the
+                hash strings of their passwords.
+            fp (string): Path to the authentication file.
+            delimiter (string): String which is used to delimit user name and
+                password hash string in the file.
+
+        Returns:
+            None
+        """
+
+        if delimiter is None:
+            delimiter = ASReviewAuth.USERPWD_DELIMITER
+
+        lines = ['%s%s%s\n' % (user, delimiter, pwd) for user, pwd in users.items()]
+        with open(fp, 'w') as f:
+            f.writelines(lines)
+
+
+    @staticmethod
+    def cmd_tool(argv):
```

Comment:
I guess it's personal, but I think the `cmd_tool()` logic might be put in `start_flask.py` so that what is passed to `auth.py` is a working auth_file.
Doing this would also allow us to move the `entry_point/auth` to the `lab` and have the main parser argument in one place without the need of an external entry point called by cmd. But this needs to be discussed with @J535D165 !

Reply:
I see your point, but would really discuss if that is a better way. Even if we make auth file manipulation a sub-command of lab cmd command, I would keep this method within dedicated auth module.